### PR TITLE
HTMLMediaElement played attribute

### PR DESCRIPTION
**Pull request by ferjm**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22093)

### CODE REVIEW

File: components/script/dom/htmlmediaelement.rs [hunk 1]
```diff
@@ -1178,7 +1183,13 @@ impl HTMLMediaElement {
                 // XXX Steps 12 and 13 require audio and video tracks support.
             },
             PlayerEvent::PositionChanged(position) => {
-                self.playback_position.set(position as f64);
+                let position = position as f64;
+                let _ = self
+                    .played
+                    .borrow_mut()
+                    .add(self.playback_position.get(), position);
+                println!("PLAYED {:?}", *self.played.borrow());
```
Comment (Manishearth) [inline components/script/dom/htmlmediaelement.rs]: remove or replace with `debug!`

File: components/script/dom/bindings/trace.rs [hunk 2]
```diff
@@ -143,6 +144,7 @@ unsafe_no_jsmanaged_fields!(&'static Encoding);
 
 unsafe_no_jsmanaged_fields!(RefCell<Decoder>);
 unsafe_no_jsmanaged_fields!(RefCell<Vec<u8>>);
+unsafe_no_jsmanaged_fields!(RefCell<TimeRangesContainer>);
```
Comment (Manishearth) [inline components/script/dom/bindings/trace.rs]: This isn't true; TimeRangesContainer is contains JS objects

Comment (Manishearth) [reply components/script/dom/bindings/trace.rs]: You probably want to use RootableVec

**Review by Manishearth — CHANGES_REQUESTED** [review]

File: components/script/dom/bindings/trace.rs (hunk 2 above)
Comment (ferjm) [reply components/script/dom/bindings/trace.rs]: @Manishearth Sorry, I am not sure how `RootableVec` would help here. [TimeRangesContainer](https://github.com/servo/servo/blob/f220a63a0f4d7757c2a5e7c9da1c2063d37bebd6/components/script/dom/timeranges.rs#L60) contains [TimeRange](https://github.com/servo/servo/blob/f220a63a0f4d7757c2a5e7c9da1c2063d37bebd6/components/script/dom/timeranges.rs#L17) objects, which IIUC are no JS objects (although they are contained by JS objects). The way I understand it [TimeRanges](https://github.com/servo/servo/blob/f220a63a0f4d7757c2a5e7c9da1c2063d37bebd6/components/script/dom/timeranges.rs#L128) (note the `s`) is the JS object here.

Comment (Manishearth) [reply components/script/dom/bindings/trace.rs]: oh I see.

Comment (jdm) [reply components/script/dom/bindings/trace.rs]: I think this line is unnecessary if you use DomRefCell instead of RefCell.

Comment (ferjm) [reply components/script/dom/bindings/trace.rs]: That's correct. I just updated the code. Thanks.

**Pull request merged**